screeps-game-api/src/raw_memory.rs

```diff
@@ -1,19 +1,16 @@
-get_from_js!(active_segments -> { Object.keys(RawMemory.segments).map(Number) } -> Vec<i32>);
-get_from_js!(get_segment(id: i32) -> { RawMemory.segments[@{id}] } -> Option<String>);
-pub fn set_segment(id: i32, data: &str) {
-    js! {
-        RawMemory.segments[@{id}] = @{data};
-    }
+//! Interface for Screeps `RawMemory` global object.
+#[derive(Deserialize, Debug)]
+pub struct ForeignSegment {
+    username: String,
+    id: String,
+    data: String
 }
-get_from_js!(foreign_segment_username -> {
-    RawMemory.forignSegment && RawMemory.forignSegment.username
-} -> Option<String>);
-get_from_js!(foreign_segment_id -> {
-    RawMemory.forignSegment && RawMemory.forignSegment.id
-} -> Option<i32>);
-get_from_js!(foreign_segment_data -> {
-    RawMemory.forignSegment && RawMemory.forignSegment.data
-} -> Option<String>);
+
+js_deserializable!(ForeignSegment);
```

Comment:
This looks nice!

I'm a bit sad about using json to transfer data back and forth but it does make sense to use it for simple stuff like this, and as you said we can tune and optimize it all later.

Reply:
Look at it as getting closer to a v1.0 and still having incentive to do better 😉